src/dataflow/src/sink/kafka.rs

```diff
@@ -394,10 +397,40 @@ impl KafkaSinkState {
         }
     }
 
-    fn send<K, P>(&self, record: BaseRecord<K, P>) -> Result<(), bool>
+    async fn init_transactions(&self, timeout: Duration) -> KafkaResult<()> {
+        let self_producer = self.producer.clone();
```

Comment:
nit: that's probably just me but I prefer the explicit `Arc::clone(&foo)` to `foo.clone()` because I never know from a glance if a `clone()` call is a real (expensive) clone or an `Rc`/`Arc` clone.

Reply:
It's also clippy lint (that we don't use), but I also use `Arc::clone(..)` for the same reason